src/routes/user/_user.paymails.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  CustomErrorComponent,
+  PaymailsTabContent,
+  Searchbar,
+  DateRangeFilter,
+  Tabs,
+  TabsContent,
+  TabsList,
+  TabsTrigger,
+  Toaster,
+} from '@/components';
+import { useSpvWalletClient } from '@/contexts';
+import { paymailsQueryOptions, addStatusField } from '@/utils';
+import { useSuspenseQuery } from '@tanstack/react-query';
+import { createFileRoute, useNavigate, useSearch } from '@tanstack/react-router';
+import { useState, useEffect } from 'react';
+
+import { useDebounce } from 'use-debounce';
+import { z } from 'zod';
+
+export const Route = createFileRoute('/user/_user/paymails')({
+  component: Paymails,
+  validateSearch: z.object({
+    sortBy: z.string().optional().catch('id'),
+    sort: z.string().optional().catch('desc'),
+    xpubId: z.string().optional().catch(''),
+    createdRange: z.object({ from: z.string(), to: z.string() }).optional().catch(undefined),
+    updatedRange: z.object({ from: z.string(), to: z.string() }).optional().catch(undefined),
+  }),
+  errorComponent: ({ error }) => <CustomErrorComponent error={error} />,
+  loader: async ({ context: { queryClient, spvWallet } }) => {
+    return await queryClient.ensureQueryData(
+      paymailsQueryOptions({
+        spvWalletClient: spvWallet.spvWalletClient!,
+      }),
+    );
+  },
+});
```

Comment:
Right now when you sort the page flickers or rerenders. It's a bad UX and in order to fix it you need to load deps from the url by doing something like this

```suggestion
  loaderDeps: ({ search: { sortBy, sort, createdRange, updatedRange } }) => ({
    sortBy,
    sort,
    createdRange,
    updatedRange,
  }),
   loader: async ({ 
      deps: { sort, sortBy, createdRange, updatedRange },
      context: { queryClient, spvWallet } 
    }) => {
    return await queryClient.ensureQueryData(
      paymailsQueryOptions({
        spvWalletClient: spvWallet.spvWalletClient!,
        sort,
        sortBy,
        createdRange,
        updatedRange,
      }),
    );
  },
```